messagingpush/src/main/java/io/customer/messagingpush/provider/FCMTokenProvider.kt

```diff
@@ -18,17 +18,22 @@ class FCMTokenProviderImpl(private val logger: Logger) : FCMTokenProvider {
     override fun getCurrentToken(onComplete: (String?) -> Unit) {
         logger.debug("getting current FCM device token...")
 
-        FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
-            if (task.isSuccessful) {
-                val existingDeviceToken = task.result
-                logger.debug("got current FCM token: $existingDeviceToken")
+        try {
```

Comment:
Why is a `try/catch` needed?

Reply:
Added the context here, 
https://github.com/customerio/issues/issues/7425

but i believe we might encounter some other issue as well so its safer to wrap it under the `try/catch` block.

Reply:
Yup, definitely a wrapper around `Firebase` may be in a similar way we have around Gist, to encapsulate its implementation and other possible crashes since FCM is something we don't have complete hold of, its always to make the whole wrapped under `try/catch`

Reply:
I added a comment to the ticket.